fabric_rti_mcp/utils/fabric_api_http_client.py

```diff
@@ -1,119 +1,190 @@
+
+        Returns:
+            Dict containing the API response
+        """
+        url = f"{self.api_base_url}{endpoint}"
+        headers = self._get_headers()
+
+        try:
+            async with httpx.AsyncClient(timeout=timeout) as client:
+                # Execute request based on method
+                if method.upper() == "GET":
+                    response = await client.get(url, headers=headers)
+                elif method.upper() == "POST":
+                    response = await client.post(url, json=payload, headers=headers)
+                elif method.upper() == "PUT":
+                    response = await client.put(url, json=payload, headers=headers)
+                elif method.upper() == "DELETE":
+                    response = await client.delete(url, headers=headers)
+                else:
+                    raise ValueError(f"Unsupported HTTP method: {method}")
+
+                # Handle response
+                if response.status_code >= 400:
+                    error_detail = response.text
+                    logger.error(f"Fabric API error {response.status_code}: {error_detail}")
+                    return {"error": True, "status_code": response.status_code, "detail": error_detail}
+
+                # Return JSON response or success message
+                if response.status_code == 204:  # No content
+                    return {"success": True, "message": "Operation completed successfully"}
+
+                try:
+                    return cast(Dict[str, Any], response.json())
+                except Exception:
+                    return {"success": True, "message": response.text}
+
+        except Exception as e:
+            logger.error(f"Error making Fabric API request: {e}")
+            return {"error": True, "message": str(e)}
+
+    def make_request(
+        self, method: str, endpoint: str, payload: Optional[Dict[str, Any]] = None, timeout: int = 30
+    ) -> Dict[str, Any]:
+        """
+        Make an authenticated HTTP request to the Fabric API (sync version).
+        
+        This is a synchronous wrapper around make_request_async that handles
+        event loop management automatically.
+
+        Args:
+            method: HTTP method (GET, POST, PUT, DELETE)
+            endpoint: API endpoint (relative to api_base_url)
+            payload: Optional request payload for POST/PUT
+            timeout: Request timeout in seconds
+
+        Returns:
+            Dict containing the API response
+        """
+        return self._run_async_operation(
+            self.make_request_async(method, endpoint, payload, timeout)
+        )
+
+
+class FabricHttpClientCache:
+    """Generic connection cache for Fabric API clients using Azure Identity."""
+
+    def __init__(self) -> None:
+        """
+        Initialize the cache.
+        """
+        self._connection: Optional[FabricAPIHttpClient] = None
+
+    def get_client(self) -> FabricAPIHttpClient:

```

Comment:
if you were to make this a `classmethod`, you'd be able to get rid of the global variable.

Reply:
done, thanks